docs/data/material/customization/css-theme-variables/usage.md

```diff
@@ -0,0 +1,394 @@
+
+If you have an existing theme, you can migrate to CSS theme variables by following the [migration guide](/material-ui/migration/migration-css-theme-variables/).
+:::
+
+## Dark mode only application
+
+To switch the default light to dark palette, set the `defaultColorScheme: 'dark'` to the `extendTheme`. Material UI will generate the dark palette instead.
+
+```jsx
+import { CssVarsProvider, extendTheme } from '@mui/material/styles';
+
+const theme = extendTheme({
+  defaultColorScheme: 'dark',
+});
+
+function App() {
+  return <CssVarsProvider theme={theme}>{/* ...you app */}</CssVarsProvider>;
+}
+```
+
+## System preferences
+
+To support both light and dark modes based on the user's system preferences, set the `cssRule: 'media'` to the `extendTheme`. Material UI will generate both light and dark palette with [`@media (prefers-color-scheme)`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme).
+
+<codeblock>
+
+```jsx JSX
+import { CssVarsProvider, extendTheme } from '@mui/material/styles';
+
+const theme = extendTheme({
+  cssRule: 'media',
+});
+
+function App() {
+  return <CssVarsProvider theme={theme}>{/* ...you app */}</CssVarsProvider>;
+}
+```
+
+```css CSS
+/* generated global stylesheet */
+
+:root {
+  --mui-palette-primary-main: #1976d2;
+  /* ...other variables */
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --mui-palette-primary-main: #90caf9;
+    /* ...other variables */
+  }
+}
+```
+
+</codeblock>
+
+<!-- TODO: fix the link -->
+
+If you want to customize the selector instead of using `prefers-color-scheme`, check out the [advanced configuration](/material-ui/customization/css-theme-variables/configuration/#advanced-configuration).
+
+## Using theme variables
+
+All of these variables are accessible in an object in the theme called `vars`.
+The structure of this object is a serializable theme structure with the values represent CSS variables.
+
+- `theme.vars` (recommended): an object that refers to the CSS theme variables.
+
+  ```js
+  const Button = styled('button')(({ theme }) => ({
+    backgroundColor: theme.vars.palette.primary.main, // var(--mui-palette-primary-main)
+    color: theme.vars.palette.primary.contrastText, // var(--mui-palette-primary-contrastText)
+  }));
+  ```
+
+  For **TypeScript**, the typings are not enabled by default.
+  Follow the [TypeScript setup](#typescript) to enable the typings.
+
+  :::success
+  If the components need to render outside of the `CssVarsProvider`, add fallback to the theme object.
+
+  ```js
+  backgroundColor: (theme.vars || theme).palette.primary.main;
+  ```
+
+  :::
+
+- **Native CSS**: if you can't access the theme object, for example in a pure CSS file, you can use [`var()`](https://developer.mozilla.org/en-US/docs/Web/CSS/var) directly:
+
+  ```css
+  /* external-scope.css */
+  .external-section {
+    background-color: var(--mui-palette-grey-50);
+  }
+  ```
+
+## Applying dark styles
```

Comment:
The Applying dark styles section should be before Using theme variables, that way all the color mode sections are together